workers/unity/Packages/com.improbable.gdk.core/Worker/Worker.cs

```
@@ -0,0 +1,85 @@
+{
+    public class Worker : IDisposable
+    {
+        public const long WorkerEntityId = -1337;
+
+        internal readonly Dictionary<EntityId, Entity> EntityMapping = new Dictionary<EntityId, Entity>();
+
+        public readonly Vector3 Origin;
+        public readonly string WorkerType;
+        public readonly string WorkerId;
+        public readonly World World;
+        public readonly Connection Connection;
+        public readonly ILogDispatcher LogDispatcher;
+
+        public delegate void WorkerEventHandler(Worker worker);
+        public static event WorkerEventHandler OnConnect;
+        public static event WorkerEventHandler OnDisconnect;
+
+        private Worker(ConnectionConfig config, ILogDispatcher logDispatcher, Vector3 origin)
+        {
+            Origin = origin;
+            WorkerId = string.IsNullOrEmpty(config.WorkerId)
+                ? $"{config.WorkerType}-{Guid.NewGuid()}"
+                : config.WorkerId;
+
+            Connection = ConnectionUtility.Connect(config, WorkerId);
+            LogDispatcher = logDispatcher;
+            WorkerType = config.WorkerType;
+            if (LogDispatcher is ForwardingDispatcher dispatcher)
+            {
+                dispatcher.SetConnection(Connection);
+            }
+
+            World = new World(WorkerId);
+            var workerSystem = World.GetOrCreateManager<WorkerSystem>();
+            workerSystem.Worker = this;
+
+            var entityManager = World.GetOrCreateManager<EntityManager>();
+            var entity = entityManager.CreateEntity();
+            entityManager.AddComponentData(entity, new OnConnected());
+            entityManager.AddComponentData(entity, new WorkerEntityTag());
+            EntityMapping.Add(new EntityId(WorkerEntityId), entity);
+            OnConnect?.Invoke(this);
+        }
+
+        public static Worker TryGetWorker(World world)
+        {
+            var workerSystem = world.GetExistingManager<WorkerSystem>();
```

If the world doesn't have a WorkerSystem it should return a proper error, instead of a NullReferenceException inside this helper function.